### PR TITLE
3651: PremiseHostingType - PUT

### PR DESCRIPTION
**Pull request by RuthBerry96**

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Solutions.API/ViewModels/Hostings/UpdateOnPremiseViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using NHSD.BuyingCatalogue.Solutions.Contracts.Commands.Hostings;
+
+namespace NHSD.BuyingCatalogue.Solutions.API.ViewModels.Hostings
+{
+    public sealed class UpdateOnPremiseViewModel : IUpdateOnPremiseData
+    {
+
```
Comment (CallumBradburyBJSS) [inline src/NHSD.BuyingCatalogue.Solutions.API/ViewModels/Hostings/UpdateOnPremiseViewModel.cs]: UNNECESSARY

File: tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/Hostings/OnPremiseHostingControllerTests.cs [hunk 2]
```diff
@@ -91,6 +105,54 @@ public async Task NullHostingShouldReturnNull()
             onPremiseResult.HostingModel.Should().BeNull();
             onPremiseResult.RequiresHSCN.Should().BeEmpty();
         }
+
+
```
Comment (CallumBradburyBJSS) [inline tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/Hostings/OnPremiseHostingControllerTests.cs]: D-D-D-DOUBLE WHITESPACE

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/Hostings/UpdateOnPremiseTests.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        private HashSet<string> _updatedRequiresHscn;
+
+        private Hosting _initialHosting;
+        private string _initialHostingJson;
+
+        [SetUp]
+        public void Setup()
+        {
+            _initialHosting = new Hosting
+
```
Comment (CallumBradburyBJSS) [inline tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/Hostings/UpdateOnPremiseTests.cs]: R-R-R-RANDOM WHITESPACE xD

File: src/NHSD.BuyingCatalogue.Solutions.Application/Commands/Hostings/OnPremise/UpdateOnPremiseValidator.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+using NHSD.BuyingCatalogue.Solutions.Application.Commands.Validation;
+
+namespace NHSD.BuyingCatalogue.Solutions.Application.Commands.Hostings.OnPremise
+{
+    class UpdateOnPremiseValidator : IValidator<UpdateOnPremiseCommand, ISimpleResult>
+    {
+        public ISimpleResult Validate(UpdateOnPremiseCommand command)
+            => new MaxLengthValidator()
+                .Validate(command.Data.Summary, 500, "summary")
```
Comment (sam-roberts-bjss) [inline src/NHSD.BuyingCatalogue.Solutions.Application/Commands/Hostings/OnPremise/UpdateOnPremiseValidator.cs]: Could data ever be null?

File: src/NHSD.BuyingCatalogue.Solutions.Application/Commands/Hostings/OnPremise/UpdateOnPremiseValidator.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+using NHSD.BuyingCatalogue.Solutions.Application.Commands.Validation;
+
+namespace NHSD.BuyingCatalogue.Solutions.Application.Commands.Hostings.OnPremise
+{
+    class UpdateOnPremiseValidator : IValidator<UpdateOnPremiseCommand, ISimpleResult>
```
Comment (sam-roberts-bjss) [inline src/NHSD.BuyingCatalogue.Solutions.Application/Commands/Hostings/OnPremise/UpdateOnPremiseValidator.cs]: internal sealed?

**Pull request merged**